File: algorithms/1_easy/M/00268_missing_number/math.rs
impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let len = nums.len();
        debug_assert!(len > 0);
        let sum = nums.iter().map(|x| *x).sum::<i32>();
        debug_assert!((len * (len + 1)) / 2 <= i32::MAX as usize);
        let target_sum = ((len * (len + 1)) / 2) as i32;

        target_sum - sum
    }
}

File: algorithms/1_easy/M/00268_missing_number/math.c
int missingNumber(int* nums, int nums_size)
{
    assert(nums != NULL && nums_size > 0);
    const ptrdiff_t len = nums_size;
    int sum = 0;

    for (ptrdiff_t i = 0; i < len; i += 1) {
        sum += nums[i];
    }

    assert((len * (len + 1)) / 2 <= INT_MAX);
    const int target_sum = (int) ((len * (len + 1)) / 2);
    return target_sum - sum;
}
